app/models/reviewable_queued_post.rb

```diff
@@ -0,0 +1,132 @@
+require_dependency 'reviewable'
+require_dependency 'user_destroyer'
+
+class ReviewableQueuedPost < Reviewable
+
+  after_create do
+    # Backwards compatibility, new code should listen for `reviewable_created`
+    DiscourseEvent.trigger(:queued_post_created, self)
+  end
+
+  def build_actions(actions, guardian, args)
+    return unless guardian.is_staff?
+
+    actions.add(:approve) unless approved?
+    actions.add(:reject) unless rejected?
+
+    if pending? && guardian.can_delete_user?(created_by)
+      actions.add(:delete_user) do |action|
+        action.icon = 'trash-alt'
+        action.label = 'reviewables.actions.delete_user.title'
+        action.confirm_message = 'reviewables.actions.delete_user.confirm'
+      end
+    end
+  end
+
+  def build_editable_fields(fields, guardian, args)
+    return unless guardian.is_staff?
+
+    # We can edit category / title if it's a new topic
+    if topic_id.blank?
+      fields.add('category_id', :category)
+      fields.add('payload.title', :text)
+      fields.add('payload.tags', :tags)
+    end
+
+    fields.add('payload.raw', :editor)
+  end
+
+  def create_options
+    result = payload.symbolize_keys
+    result[:cooking_options].symbolize_keys! if result[:cooking_options]
+    result[:topic_id] = topic_id if topic_id
+    result[:category] = category_id if category_id
+    result
+  end
+
+  def perform_approve(performed_by, args)
+    created_post = nil
+
+    creator = PostCreator.new(created_by, create_options.merge(
+      skip_validations: true,
+      skip_jobs: true,
+      skip_events: true
+    ))
+    created_post = creator.create
+
+    unless created_post && creator.errors.blank?
+      return create_result(:failure) { |r| r.errors = creator.errors }
+    end
+
+    payload['created_post_id'] = created_post.id
+    payload['created_topic_id'] = created_post.topic_id unless topic_id
+    save
+
+    UserSilencer.unsilence(created_by, performed_by) if created_by.silenced?
+
+    StaffActionLogger.new(performed_by).log_post_approved(created_post) if performed_by.staff?
+
+    # Backwards compatibility, new code should listen for `reviewable_transitioned_to`
+    DiscourseEvent.trigger(:approved_post, self, created_post)
+
+    create_result(:success, :approved) { |result| result.created_post = created_post }
+  end
+
+  def perform_reject(performed_by, args)
+    # Backwards compatibility, new code should listen for `reviewable_transitioned_to`
+    DiscourseEvent.trigger(:rejected_post, self)
+
+    StaffActionLogger.new(performed_by).log_post_rejected(self, DateTime.now) if performed_by.staff?
+
+    create_result(:success, :rejected)
+  end
+
+  def perform_delete_user(performed_by, args)
+    delete_options = {
+      context: I18n.t('reviewables.actions.delete_user.reason'),
+      delete_posts: true,
+      delete_as_spammer: true
+    }
+
+    if Rails.env.production? && ENV["Staging"].nil?
```

Comment:
I'm curious about this conditional here. Is there a specific use case for this? My worry here is that the ENV is very generic and its use case isn't documented.

Reply:
This isn't new code, it's old Queued Posts code that I migrated over. I looked back and I wrote that code many years ago. It's possible we don't need it but it's probably safer to leave for now.

Reply:
Ah I’m quite sure we can nuke this code now. I don’t think we have any clusters with the Staging ENV. Plus, I would consider this a private API so we don’t have to worry about deprecating.

Reply:
Sure thing, nuked in an upcoming commit.